services/blockchain-indexer/shared/utils/validatorUtils.js

```diff
@@ -51,7 +68,36 @@ const getAddressByName = async (name) => {
 	return null;
 };
 
+const calcCommission = async (generatorAddress, reward) => {
+	const commissionsTable = await getCommissionsTable();
+	const [{ commission: currentCommission }] = await commissionsTable
+		.find({ address: generatorAddress, sort: 'height:desc', limit: 1 }, 'commission');
+	const commission = (BigInt(reward) * BigInt(currentCommission)) / MAX_COMMISSION;
```

Comment:
We have to make sure that the result of such floating number division is same as the one produced in the protocol implementation (which uses Q96 numbers)